spec/mock-request.js

```diff
@@ -57,23 +78,27 @@ HttpBackend.prototype = {
                 flushed += 1;
                 if (numToFlush && flushed === numToFlush) {
                     console.log("  [%s] Flushed assigned amount: %s", path, numToFlush);
+                    // Restore Clock
+                    if (self.useMockClock) {
+                        callbacks.setNow();
+                    }
                     defer.resolve();
                 } else {
-                    setTimeout(tryFlush, 0);
+                    setImmediate(tryFlush);
                 }
-            } else if (flushed === 0 && !triedWaiting) {
-                // we may not have made the request yet, wait a generous amount of
-                // time before giving up.
-                setTimeout(tryFlush, 5);
-                triedWaiting = true;
+            } else if (flushed === 0 && waitLoops < 50) {
+                waitLoops += 1
```

Comment:
`setTimeout` for waiting isn't compatible with mocking that out, so I ended up with something like this -- but it seems like a hack. I believe Jasmine has a waitFor (but it's removed in 2.0) it would be great to do this in a nicer way though...